dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/service/impl/TaskDefinitionServiceImpl.java

```diff
@@ -97,28 +100,53 @@
             return result;
         }
 
-        TaskNode taskNode = JSONUtils.parseObject(taskDefinitionJson, TaskNode.class);
-        checkTaskNode(result, taskNode, taskDefinitionJson);
-        if (result.get(Constants.STATUS) == DATA_IS_NOT_VALID
-                || result.get(Constants.STATUS) == Status.PROCESS_NODE_S_PARAMETER_INVALID) {
-            return result;
+        List<TaskDefinition> taskDefinitions = JSONUtils.toList(taskDefinitionJson, TaskDefinition.class);
+        int totalSuccessNumber = 0;
+        List<Long> totalSuccessCode = new ArrayList<>();
+        List<TaskDefinition> taskDefinitionList = new ArrayList<>();
+        List<TaskDefinitionLog> taskDefinitionLogsList = new ArrayList<>();
+        for (TaskDefinition taskDefinition : taskDefinitions) {
+            checkTaskDefinition(result, taskDefinition, taskDefinitionJson);
+            if (result.get(Constants.STATUS) == DATA_IS_NOT_VALID
+                    || result.get(Constants.STATUS) == Status.PROCESS_NODE_S_PARAMETER_INVALID) {
+                return result;
+            }
+            taskDefinition.setProjectCode(projectCode);
+            taskDefinition.setUserId(loginUser.getId());
+            taskDefinition.setVersion(1);
+            Date now = new Date();
+            taskDefinition.setCreateTime(now);
+            taskDefinition.setUpdateTime(now);
+            long code = 0L;
+            try {
+                code = SnowFlakeUtils.getInstance().nextId();
+                taskDefinition.setCode(code);
+            } catch (SnowFlakeException e) {
+                logger.error("Task code get error, ", e);
+            }
+            if (code == 0L) {
```

Comment:
It's unnecessary to judge the code here. You can put the if code block into the catch